plugins/modules/azure_rm_containerregistrytag_info.py

```diff
@@ -0,0 +1,235 @@
+
+try:
+    from azure.containerregistry import ContainerRegistryClient
+except ImportError as exc:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMContainerRegistryTagInfo(AzureRMModuleBase):
+    def __init__(self):
+        self.module_arg_spec = dict(
+            registry=dict(
+                type="str",
+                required=True,
+            ),
+            repository_name=dict(
+                type="str",
+            ),
+            name=dict(
+                type="str",
+            ),
+        )
+
+        self.results = dict(
+            changed=False
+        )
+
+        self.registry = None
+        self.repository_name = None
+        self.name = None
+
+        self._client = None
+
+        super(AzureRMContainerRegistryTagInfo, self).__init__(self.module_arg_spec,
+                                                              supports_check_mode=True,
+                                                              supports_tags=False,
+                                                              facts_module=True)
+
+    def exec_module(self, **kwargs):
+        for key in list(self.module_arg_spec.keys()):
+            setattr(self, key, kwargs[key])
+
+        self._client = self.get_client()
+
+        if self.repository_name and self.name:
+            self.results["repositories"] = [self.get_tag(self.repository_name, self.name)]
+        elif self.repository_name:
+            self.results["repositories"] = [self.list_by_repository(self.repository_name, self.name)]
+        else:
+            self.results["repositories"] = self.list_all_repositories(self.name)
+
+        return self.results
+
+    def get_client(self):
+        registry_endpoint = self.registry if self.registry.endswith(".azurecr.io") else self.registry + ".azurecr.io"
+        return ContainerRegistryClient(
+            endpoint=registry_endpoint,
+            credential=self.azure_auth.azure_credential_track2,
+            audience="https://management.azure.com",
+        )
+
+    def get_tag(self, repository_name, tag_name):
+        response = None
+        try:
+            response = self._client.get_tag_properties(repository=repository_name, tag=tag_name)
+            self.log(f"Response : {response}")
+        except Exception as e:
+            self.log(f"Could not get ACR tag for {repository_name}:{tag_name} - {str(e)}")
+
+        tags = []
+        if response is not None:
+            tags.append(format_tag(response))
+
+        return {
+            "name": repository_name,
+            "tags": tags,
+        }
+
+    def list_by_repository(self, repository_name, tag_name):
+        response = None
+        try:
+            response = self._client.list_tag_properties(repository=repository_name)
+            self.log(f"Response : {response}")
+        except Exception as e:
+            self.log(f"Could not get ACR tag for {repository_name} - {str(e)}")
+
+        if response is not None:
```

Comment:
Can you please elaborate? There is exception handling for the API call and the looping logic will not run because `response` is not set to a value due to the exception. So I think the flow is working as expected. Thanks!

Reply:
Thank you for the detail! I have pushed a fix and added a test case in https://github.com/ansible-collections/azure/pull/830/commits/103ce7aef0633e01a42a1da727162761044cd67d.